File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario1/id_13T2000Test.py
# id_13_Test.java
from scenario1.id_13 import greatest_common_divisor

import pytest

class id_13_Test:
    """
    Test class of id_13.py.
    It contains ten unit test cases for the :func:`scenario1.id_13.greatest_common_divisor`.
    """


    def test_greatest_common_divisor_1(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(1, 1) == 1

    def test_greatest_common_divisor_2(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(1, 2) == 1

    def test_greatest_common_divisor_3(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(2, 1) == 1

    def test_greatest_common_divisor_4(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(2, 2) == 2

    def test_greatest_common_divisor_5(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(2, 3) == 1

    def test_greatest_common_divisor_6(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(3, 2) == 1

    def test_greatest_common_divisor_7(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(3, 3) == 3

    def test_greatest_common_divisor_8(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(3, 4) == 1

    def test_greatest_common_divisor_9(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(4, 3) == 1

    def test_greatest_common_divisor_10(self):
        """
        Test case for the :func:`scenario1.id_13.greatest_common_divisor`.
        """
        assert greatest_common_divisor(4, 4) == 4

File: RQ2_Prompt_Elements/StarCoder_Data/HumanEvalJava_output/scenario1/id_13T2000Test.java
// GreatestCommonDivisorTest.java
package scenario1;


import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link GreatestCommonDivisor}.
* It contains ten unit test cases for the {@link GreatestCommonDivisor#greatestCommonDivisor(int, int)} method.
*/
class GreatestCommonDivisorTest {

    @Test
    void testGreatestCommonDivisor1() {
        assertEquals(1, GreatestCommonDivisor.greatestCommonDivisor(1, 1));
    }

    @Test
    void testGreatestCommonDivisor2() {
        assertEquals(1, GreatestCommonDivisor.greatestCommonDivisor(1, 2));
    }

    @Test
    void testGreatestCommonDivisor3() {
        assertEquals(1, GreatestCommonDivisor.greatestCommonDivisor(2, 1));
    }

    @Test
    void testGreatestCommonDivisor4() {
        assertEquals(2, GreatestCommonDivisor.greatestCommonDivisor(2, 2));
    }

    @Test
    void testGreatestCommonDivisor5() {
        assertEquals(1, GreatestCommonDivisor.greatestCommonDivisor(1, 3));
    }

    @Test
    void testGreatestCommonDivisor6() {
        assertEquals(1, GreatestCommonDivisor.greatestCommonDivisor(3, 1));
    }

    @Test
    void testGreatestCommonDivisor7() {
        assertEquals(3, GreatestCommonDivisor.greatestCommonDivisor(3, 3));
    }

    @Test
    void testGreatestCommonDivisor8() {
        assertEquals(1, GreatestCommonDivisor.greatestCommonDivisor(1, 4));
    }

    @Test
    void testGreatestCommonDivisor9() {
        assertEquals(1, GreatestCommonDivisor.greatestCommonDivisor(4, 1));
    }

    @Test
    void testGreatestCommonDivisor10() {
        assertEquals(4, GreatestCommonDivisor.greatestCommonDivisor(4, 4));
    }
}
